### DISCUSSION TITLE
How do I bundle a service worker, as suggested in the guide "Session management with service workers"

### DISCUSSION THREAD

**Discussion by ben519** [firebase/firebase-js-sdk#8458] [Q&A]
I'm following the guide [Session management with service workers](https://firebase.google.com/docs/auth/web/service-worker-sessions). Currently I'm getting the error

> Cannot use import statement outside a module (at service-worker.js:1:1)

I suspect it's related to this note in the guide

> Make sure that the service worker is bundled so that it will still work after the browser has been closed.

Can anyone guide me on how to bundle a service worker using Next.js? This feels like a significant step that is glossed over, and I'm quite lost.

Thanks!

**Comment by raky291** [comment]
I agree, Firebase team should improve the documentation for Next.js
